app/src/main/scala/org/alephium/explorer/persistence/queries/BlockQueries.scala

```diff
@@ -311,4 +313,63 @@ object BlockQueries extends StrictLogging {
       ORDER BY block_timestamp
     """.as[TimeStamp]
   }
+
+  /**
+    * Fetches the maximum `block_timestamp` from blocks
+    * with maximum height within the given chain.
+    */
+  val maxBlockTimestampForMaxHeightForChainSQL: String =
+    s"""
+       |SELECT max(block_timestamp) as block_timestamp,
+       |       height
+       |FROM $block_headers
+       |WHERE height = (SELECT max(height)
+       |                FROM $block_headers
+       |                WHERE chain_from = ?
+       |                  AND chain_to = ?)
+       |  AND chain_from = ?
+       |  AND chain_to = ?
+       |GROUP BY height
+       |""".stripMargin
+
+  /**
+    * Fetches the maximum `block_timestamp` and sum height of
+    * all blocks filtered with maximum height for all input chain
+    * chains provided by [[GroupSetting.groupIndexes]]
+    *
+    * @param groupSetting Provides the list of chains to run this query on
+    * @return Maximum timestamp and sum of all heights i.e. the total number of blocks.
+    */
+  def noOfBlocksAndMaxBlockTimestamp()(
```

Comment:
super minor: I spent few seconds understanding why the function start with `no` (likethe contrary of `yes`.... ), maybe we could use `numOfBlocksAndMaxBlockTimestamp` ?